### PR TITLE
Revert from alpine to phusion and include wasm build

### PR DESCRIPTION
**Pull request by chevdor**
Alpine base images make great docker images because they are very small. Some of the binaries come unfortunately with delay. It happened already several times for the Substrate and Polkadot repos that the cargo version available on alpine was a tad too old and did not allow building substrate/polkadot.

The docker image for polkadot was reverted a while back already.

This PR reverts from `alpine` to using `phusion/base`. I did attempt to keep alpine as second stage which ends up being a little headache. The reason for that is that alpine uses `musl` instead of `glibc`. So the options would be to build using `musl` or install `glibc` in alpine. The second option is not trivial and ideal as the size will grow significantly.

I also tried a second stage using debian:stretch but the (not)included libssl libs make it a mess.
Thus my suggestion to stick with `phusion/baseimage` until both substrate and cargo stabilize.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,38 +1,52 @@
...
 WORKDIR /substrate
 
 COPY . /substrate
 
-RUN cargo build --$PROFILE
+RUN apt-get update && \
+	apt-get upgrade -y && \
+	apt-get install -y cmake pkg-config libssl-dev git clang
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
```
Comment (DemiMarie-parity) [inline Dockerfile]: We might want to download it ourselves, so that we can check the signatures.

File: Dockerfile [hunk 2]
```diff
@@ -1,38 +1,52 @@
...
+	cargo install --git https://github.com/alexcrichton/wasm-gc && \
+	rustup default nightly && \
+ 	./scripts/build.sh && \
+ 	rustup default stable && \
+	cargo build --$PROFILE
 
 # ===== SECOND STAGE ======
 
-FROM alpine:edge
+FROM phusion/baseimage:0.10.2
```
Comment (DemiMarie-parity) [inline Dockerfile]: Could we build against musl instead, statically link everything, and then do the second stage from Alpine, or even better `FROM scratch`?

Comment (chevdor) [reply Dockerfile]: That would be very cool indeed. I do lack time atm however and I wanted to first have a working solution. We can always build up from that point on.

File: Dockerfile (hunk 1 above)
Comment (chevdor) [reply Dockerfile]: Can you provide details? I am not sure what you have in mind. You want to download Rust manually? Against what would you check the signatures?

**Comment by chevdor** [comment]
> Would it be possible to include comments explaining why the chosen approach was used?

Hi @DemiMarie-parity, I think I did explain in the PR description. If you think there is a point to explain with more details, give me a hint on what is not clear and I'll be glad to expand on the topic.

**Pull request merged**